spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/eventhub/AzureEventHubAutoConfiguration.java

```diff
@@ -7,30 +7,71 @@
+import com.microsoft.azure.management.eventhub.EventHubAuthorizationKey;
+import com.microsoft.azure.management.eventhub.EventHubNamespace;
 import com.microsoft.azure.spring.cloud.autoconfigure.context.AzureContextAutoConfiguration;
 import com.microsoft.azure.spring.cloud.autoconfigure.context.AzureProperties;
-import eventhub.integration.AzureAdmin;
+import com.microsoft.azure.spring.cloud.context.core.AzureAdmin;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.kafka.KafkaAutoConfiguration;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import java.util.Arrays;
 
 /**
- * An auto-configuration for Event Hub
+ * An auto-configuration for Event Hub, which provides {@link KafkaProperties}
  *
  * @author Warren Zhu
  */
 @Configuration
+@AutoConfigureBefore(KafkaAutoConfiguration.class)
 @AutoConfigureAfter(AzureContextAutoConfiguration.class)
 @ConditionalOnClass(EventHub.class)
+@ConditionalOnProperty("spring.cloud.azure.eventhub.namespace")
 @EnableConfigurationProperties(AzureEventHubProperties.class)
 public class AzureEventHubAutoConfiguration {
+    private static final String SECURITY_PROTOCOL = "security.protocol";
+    private static final String SASL_SSL = "SASL_SSL";
+    private static final String SASL_JAAS_CONFIG = "sasl.jaas.config";
+    private static final String SASL_CONFIG_VALUE =
+            "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" " +
+                    "password=\"%s\";\n";
+    private static final String SASL_MECHANISM = "sasl.mechanism";
+    private static final String SASL_MECHANISM_PLAIN = "PLAIN";
+    private static final int PORT = 9093;
 
     @Bean
-    public AzureAdmin azureAdmin(Azure azure, AzureProperties azureProperties){
+    public AzureAdmin azureAdmin(Azure azure, AzureProperties azureProperties) {
         return new AzureAdmin(azure, azureProperties.getResourceGroup(), azureProperties.getRegion());
     }
+
+    @ConditionalOnMissingBean
+    @Primary
+    @Bean
+    public KafkaProperties kafkaProperties(AzureAdmin azureAdmin, AzureEventHubProperties eventHubProperties) {
+        KafkaProperties kafkaProperties = new KafkaProperties();
+        EventHubNamespace namespace = azureAdmin.getOrCreateEventHubNamespace(eventHubProperties.getNamespace());
```

Comment:
Is it possible azureAdmin hasn't been created yet?

Reply:
AzureAdmin is autowired as bean in previous method. Spring should handle such dependency correctly.